src/schema/duration.rs

```diff
@@ -0,0 +1,358 @@
+                }
+
+                // Duration components:
+                'Y' => {
+                    fill_component(&mut context, &mut dur.years, 1, "years", 'Y')?;
+                }
+                'M' => {
+                    if context.t_found {
+                        fill_component(&mut context, &mut dur.minutes, 5, "minutes", 'M')?;
+                    } else {
+                        fill_component(&mut context, &mut dur.months, 2, "months", 'M')?;
+                    }
+                }
+                'D' => {
+                    fill_component(&mut context, &mut dur.days, 3, "days", 'D')?;
+                }
+                'H' => {
+                    if !context.t_found {
+                        return Err("No symbol 'T' found before hours components");
+                    }
+                    fill_component(&mut context, &mut dur.hours, 4, "hours", 'H')?;
+                }
+                'S' => {
+                    if !context.t_found {
+                        return Err("No symbol 'T' found before seconds components");
+                    }
+                    fill_seconds(&mut context, &mut dur.seconds)?;
+                }
+
+                // Number:
+                '.' => {
+                    if context.dot_found {
+                        return Err("Dot occurred twice");
+                    }
+
+                    if context.number_is_empty {
+                        return Err("No digits before dot");
+                    }
+
+                    context.dot_found = true;
+                }
+                digit => {
+                    if !digit.is_digit(10) {
+                        return Err("Incorrect character occurred");
+                    }
+
+                    if context.dot_found {
+                        context.numer *= 10;
+                        context.numer +=
+                            digit.to_digit(10).expect("error converting a digit") as u64;
+                        context.denom *= 10;
+                    } else {
+                        context.number *= 10;
+                        context.number +=
+                            digit.to_digit(10).expect("error converting a digit") as u64;
+                        context.number_is_empty = false;
+                    }
+                }
+            }
+        }
+
+        if context.number > 0 {
+            return Err("Number at the end of the string");
+        }
+
+        if !context.p_found {
+            return Err("'P' must always be present");
+        }
+
+        if context.last_filled_component == 0 {
+            return Err("At least one number and designator are required");
+        }
+
+        if context.last_filled_component <= 3 && context.t_found {
+            return Err("No time items are present, so 'T' must not be present");
+        }
+
+        Ok(dur)
+    }
+}
+
+struct ParsingContext {
+    p_found: bool,              // Is 'P' found in the string.
```

Comment:
To be consistent with other `bool` variables, either use `is_p_found` or remove `is` from other bools.

Reply:
Thanks, fixed. Used `is_` for all `bool` fields.